TST: Allow for multiple variables on the same line in docstring validation

ChiefMilesEdgeworth opened the pull request:
Allow for multiple variables to share the same type and description in documentation. For example, `i, j, k : int` Would be three variables, i, j, and k, which are all ints.

- [ ] tests added / passed 
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry - I'm not sure how to add this, if it needs done could I get a pointer?

ChiefMilesEdgeworth committed:
TST: Allow for multiple variables on the same line in doctests

Allow for multiple variables to share the same type and description in documentation. For example, `i, j, k : int` Would be three variables, i, j, and k, which are all ints.

ChiefMilesEdgeworth committed:
DOC: Fix order in series.py

This was generating errors because it got fixed in master after I made the branch. I'm sure this will be a merge conflict, and I can work it out there first.

WillAyd reviewed (changes requested):
Can you add test(s) for this change? There is a test directory next to the validation script you can put those in

ChiefMilesEdgeworth committed:
Added test

Added a test for if the spacing is off on a multivariable oneline parameter descriptoin

Nabel0721 commented:
> Can you add test(s) for this change? There is a test directory next to the validation script you can put those in

I put in a test for if spacing is off. Other than that, it's a little hard to test because I can't have a test with more than one wrong parameter. The validation script uses sets, so the order can change between runs.

WillAyd commented:
Can you not put a test in `class GoodDocStrings` that has this behavior to make sure it works?

Nabel0721 commented:
Oh, yeah. I'll do that real quick. Sorry, didn't cross my mind.

ChiefMilesEdgeworth committed:
TST: Correct test for multiple variables on one line

Adds a test for if multiple variables are typed and described on the same line in a docstring.

WillAyd commented on scripts/validate_docstrings.py:
```diff
@@ -422,10 +422,11 @@ def needs_summary(self):
-        )
+        parameters = collections.OrderedDict()
+        for names, type_, desc in self.doc["Parameters"]:
+            for name in names.split(", "):
```
Can you just do a `re.findall` here instead? I don't think whitespace is important

datapythonista replied:
I'd prefer that we're consistent with the spacing after the comma. I think it's a bit distracting having a mix of `foo, bar`, `foo,bar`, `foo ,bar`...

In a follow up could be nice to have more specific error messages when the problem is the spacing, but this looks good to me for this PR.

WillAyd replied:
OK sounds good. Would be nice to document that somewhere if it isn't already then (follow up PR is fine)

WillAyd replied:
Also not sure if that needs its own error code

WillAyd reviewed (changes requested):
minor comment otherwise lgtm @datapythonista if you have any thouhts

datapythonista commented on scripts/tests/test_validate_docstrings.py:
```diff
@@ -39,6 +39,23 @@ def plot(self, kind, color="blue", **kwargs):
 
+    def swap(self, arr, i, j, *args, **kwargs):
+        """
+        Swap two indecies on an array.
```
```suggestion
        Swap two indices on an array.
```

datapythonista commented on scripts/tests/test_validate_docstrings.py:
```diff
@@ -39,6 +39,23 @@ def plot(self, kind, color="blue", **kwargs):
+
+        Parameters
+        ----------
+        arr : List
```
```suggestion
        arr : list```

datapythonista commented on scripts/tests/test_validate_docstrings.py:
```diff
@@ -39,6 +39,23 @@ def plot(self, kind, color="blue", **kwargs):
+        """
+        temp = arr[i]
+        arr[i] = arr[j]
+        arr[j] = temp
```
if we don't use this code, better have just `pass`

datapythonista commented on scripts/tests/test_validate_docstrings.py:
```diff
@@ -256,6 +273,23 @@ def say_hello():
+        """
+        temp = matrix[a][b]
+        matrix[a][b] = matrix[i][j]
+        matrix[i][j] = temp
```
same, replace with `pass`

datapythonista reviewed (commented):
nice PR, just couple of typos, but this is very useful

ChiefMilesEdgeworth committed:
TST: Change tests to just pass, fix typos

Removed the functionality of tests, since it isn't important for documentation. Fixed a couple of small typos.

WillAyd reviewed (approved):
OK by me - merge whenever happy @datapythonista

jorisvandenbossche reviewed (approved):
Looks good to me as well, thanks a lot for the contribution!

WillAyd commented:
Thanks @Nabel0721 !
